client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/RssShuffleManagerBase.java

```diff
@@ -36,34 +37,70 @@
 import org.apache.spark.shuffle.SparkVersionUtils;
+import org.apache.spark.shuffle.handle.MutableShuffleHandleInfo;
+import org.apache.spark.shuffle.handle.ShuffleHandleInfo;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import org.apache.uniffle.client.api.CoordinatorClient;
+import org.apache.uniffle.client.api.ShuffleManagerClient;
+import org.apache.uniffle.client.api.ShuffleWriteClient;
 import org.apache.uniffle.client.factory.CoordinatorClientFactory;
 import org.apache.uniffle.client.request.RssFetchClientConfRequest;
 import org.apache.uniffle.client.response.RssFetchClientConfResponse;
+import org.apache.uniffle.client.util.ClientUtils;
 import org.apache.uniffle.common.ClientType;
+import org.apache.uniffle.common.PartitionRange;
 import org.apache.uniffle.common.RemoteStorageInfo;
+import org.apache.uniffle.common.ShuffleAssignmentsInfo;
+import org.apache.uniffle.common.ShuffleDataDistributionType;
+import org.apache.uniffle.common.ShuffleServerInfo;
 import org.apache.uniffle.common.config.ConfigOption;
 import org.apache.uniffle.common.config.RssClientConf;
 import org.apache.uniffle.common.config.RssConf;
 import org.apache.uniffle.common.exception.RssException;
 import org.apache.uniffle.common.rpc.StatusCode;
+import org.apache.uniffle.common.util.RetryUtils;
 import org.apache.uniffle.shuffle.BlockIdManager;
 
 import static org.apache.uniffle.common.config.RssClientConf.HADOOP_CONFIG_KEY_PREFIX;
 import static org.apache.uniffle.common.config.RssClientConf.RSS_CLIENT_REMOTE_STORAGE_USE_LOCAL_CONF_ENABLED;
 
-public abstract class RssShuffleManagerBase implements RssShuffleManagerInterface, ShuffleManager {
+public abstract class RssShuffleManagerBase
+    implements RssShuffleManagerInterface, RssShuffleManagerRetryInterface, ShuffleManager {
   private static final Logger LOG = LoggerFactory.getLogger(RssShuffleManagerBase.class);
   private AtomicBoolean isInitialized = new AtomicBoolean(false);
   private Method unregisterAllMapOutputMethod;
   private Method registerShuffleMethod;
   private volatile BlockIdManager blockIdManager;
   private Object blockIdManagerLock = new Object();
+  protected String appId = "";
+  protected ShuffleWriteClient shuffleWriteClient;
+  protected boolean dynamicConfEnabled;
+  protected int maxConcurrencyPerPartitionToWrite;
+  protected String clientType;
+
+  protected SparkConf sparkConf;
+  protected ShuffleManagerClient shuffleManagerClient;
+  /** Whether to enable the dynamic shuffleServer function rewrite and reread functions */
+  protected boolean rssResubmitStage;
+  /**
+   * Mapping between ShuffleId and ShuffleServer list. ShuffleServer list is dynamically allocated.
+   * ShuffleServer is not obtained from RssShuffleHandle, but from this mapping.
+   */
+  protected ShuffleHandleInfoManager shuffleHandleInfoManager;
+  /** A list of shuffleServer for Write failures */
+  protected Set<String> failuresShuffleServerIds;
```

Comment:
I hope the stage resubmit related vars could be managed in one independent class, like `RssStageResubmitManager` ?